lessons/babel.md

```diff
@@ -1 +1,130 @@
+  - You can pass arguments to babel through the command line, or the Node API but having them specified in an rc file, project-wide means less repetition if you're have other libraries in your project that use babel and you want the same setup (e.g. `babel-jest` or `webpack`)
+- For now we'll just need a file that tells babel what plugins we're using in a json format - the biggest use case for a `.babelrc` file:
+```json
+{
+  "presets": ["@babel/env"],
+  "plugins": ["@babel/plugin-proposal-object-rest-spread"]
+}
+```
+- Before this will work we actually need to install those plugins with `yarn add --dev @babel/env "@babel/plugin-proposal-object-rest-spread`
+
+#### Aside: Presets and plugins
+"Presets" are just a packaged list of "plugins"
+
+##### Preset env
+Preset env is a "special" preset that contains all "stage-4" proposals from the TC39 committee - the committee that decides on the ECMA spec (the one that Javascript follows)
+
+- stage-0 - Strawman
+- stage-1 - Proposal (has a champion)
+- stage-2 - Draft
+- stage-3 - Candidate
+- stage-4 - Finished (will be added to the spec, waiting to be implemented in runtimes)
+
+##### Object rest spread
+Very useful and is now stage-4 although not moved to stage 4 in babel so we've added it here as a good example of a plugin
+
+### Running Babel
+Installing `babel-cli` gave us a file in `node_modules/.bin` called `babel` that acts as the babel CLI (this pattern is nice as node can install all binary dependencies without relying on the environment to provide them). We can call this file `$(yarn bin)/babel` (`$(yarn bin)` gives us the absolute path to this bin file - try typing `yarn bin` into the command line). We only need to specify two things:
+- A directory (or file) to run it on (the first command line argument)
+- An output directory (or file) - specfied with the `-d` or `-o` (for file) flag.
+
+All of these things together mean we can run `mkdir -p dist && $(yarn bin)/babel src -d dist`.
+
+### Yarn / npm scripts
+To save us typing this each time we can instead add a `script` inside our package.json like so (these scripts automatically get `node_modules/.bin` added to their `$PATH` so we can remove `$(yarn bin)/`:
+
+```js
+  {
+    //...
+    "scripts": {
+      "babel": "mkdir -p dist && babel src -d dist"
+    }
+    // ...
+  }
+```
+
+We now just need to change our `index.html` file to load these `dist` files rather than our `src` files. If you load the browser then hopefully nothing should have changed!
+
+### Polyfilling
+Finally, even though we porbably won't see any errors (even in Safari), as mentioned above we will need to polyfill missing functions from later browsers.
+
+If you look in `dist/services/CAPI.js` you'll see references to `regeneratorRuntime` which babel needs to support `async / await` in older browsers. As mentioned earlier this requires polyfilling.
```

Comment:
`Promise` was mentioned earlier but not async/await specifically. Maybe we could change the earlier example